doublyLinkedLists.js

```diff
@@ -0,0 +1,499 @@
+// function constructor for doubly linked lists
+function DoublyLinkedList() {
+    
+    this._size = 0;
+    this.head = null;
+    this.tail = null;
+    
+}
+
+DoublyLinkedList.prototype = {
+    
+    Node: function (data) {
+        
+        this.previous = null;
+        this.data = data;
+        this.next = null;
+        
+    },
+    
+    // insert new node at the front of the list
+    insertFirst: function (data) {
+        
+        var current = this.head,
+            newNode = new this.Node(data);
+        
+        
+        // if list is empty set head and tail equal to newNode
+        if (this._size === 0) {
+                
+            this.tail = this.head = newNode;
+            
+        } else {
+            
+            // set head equal to newNode
+            newNode.next = current;
+            current.previous = newNode;
+            this.head = newNode;
+            
+        }
+        
+        this._size++;
+        
+        return newNode;
+    
+    },
+    
+    // delete first node in the list
+    deleteFirst: function () {
+        
+        var deleted;
+        
+        // if list contains at least one node
+        if (this._size > 0) {
+            
+            // head will be deleted
+            deleted = this.head;
+            
+            // if list contains only one node, delete head and tail
+            if (this._size === 1) {
+                
+                this.head = this.tail = null;
+                this._size--;
+                
+                return deleted;
+                
+            }
+            
+            // delete head and set new head equal to the following node
+            this.head = this.head.next;
+            this.head.previous = null;
+            this._size--;
+            
+            return deleted;
+            
+        }
+        
+        // throw an error if list is empty
+        throw new Error('No node to delete');
+        
+    },
+    
+    // insert node at the end of the list
+    insertLast: function (data) {
```

Comment:
I have trouble with this naming convention. Head is the first or last? Maybe there is a better way? some people use left and right, some first and last (python deque)[https://docs.python.org/2/library/collections.html#collections.deque]